src/main/java/org/truffleruby/core/method/MethodNodes.java

```diff
@@ -409,4 +409,13 @@ protected Object allocate(RubyClass rubyClass) {
 
     }
 
+    @Primitive(name = "basic_method?")
+    public abstract static class BasicMethodNode extends PrimitiveArrayArgumentsNode {
+
+        @Specialization
+        protected boolean basicMethod(RubyMethod method) {
+            return method.method.getSharedMethodInfo().isBasicMethod;
```

Comment:
We already have `InternalMethod#builtIn` and `Primtive.vm_method_is_basic(someMethod)` that seem very similar.